internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -253,14 +256,42 @@ func (r *ClusterDeploymentsReconciler) Reconcile(origCtx context.Context, req ct
 	return r.updateStatus(ctx, log, clusterInstall, clusterDeployment, cluster, nil)
 }
 
-func (r *ClusterDeploymentsReconciler) validateClusterDeployment(clusterDeployment *hivev1.ClusterDeployment,
+func (r *ClusterDeploymentsReconciler) validateClusterDeployment(ctx context.Context, clusterDeployment *hivev1.ClusterDeployment,
 	clusterInstall *hiveext.AgentClusterInstall) error {
 
 	// Make sure that the ImageSetRef is set for clusters not already installed
 	if clusterInstall.Spec.ImageSetRef == nil && !clusterDeployment.Spec.Installed {
 		return newInputError("missing ImageSetRef for cluster that is not installed")
 	}
 
+	// If mirror registries are enabled then we need to make sure that the release image is not tag based.
+	// A tagged release image will be rejected by oc when mirror registry configuration is present.
+	// Mirrored images are typically used in disconnected environments which can have trouble with tag based images.
+	hasMirrorRegistries, err := r.MirrorRegistriesConfigBuilder.HasMirrorRegistries()
+	if err != nil {
+		return errors.Wrap(err, "unable to validate release image due to error determining mirror registry status")
+	}
+	if hasMirrorRegistries && clusterInstall.Spec.ImageSetRef != nil {
+		clusterImageSet := &hivev1.ClusterImageSet{}
```

Comment:
We have this code in other place, maybe lets create function for it ?

Reply:
True that it's very similar to getReleaseImage but as the purpose of this fix is to cater for an issue affecting disconnected environments, there is a possibility that this will not work.

`func (r *ClusterDeploymentsReconciler) getReleaseImage( `

Makes an internal call to fetch further repository information, this is overkill for my use case as I only need the URL.
Also, for the problem I am trying to solve, I feel it will lead to misleading errors that do not advise the user as intended.

https://github.com/openshift/assisted-service/blob/9462a1af97cd3efb009894ce6dd0b75da06e65b9/internal/versions/kube_api_versions.go#L111

Reply:
Done, I refactored.